File: 220818/JavaScript/weird-string.js
export default function solution(string) {
  let weirdString = '';

  let wordCount = 0;

  for (let i = 0; i < string.length; i += 1) {
    if (string[i] === ' ') {
      weirdString += ' ';
      wordCount = 0;
      continue;
    }

    if (wordCount % 2 === 0) {
      weirdString += string[i].toUpperCase();
      wordCount += 1;
      continue;
    }

    if (wordCount % 2 === 1) {
      weirdString += string[i].toLowerCase();
      wordCount += 1;
      continue;
    }
  }

  return weirdString;
}

console.log(solution('try hello world'));

File: solutions/Previous/220818/Java/WeirdString.java
public class WeirdString {
  public String solution(String normalString) {
    String[] weirdString = new String[normalString.length()];
    int wordIndex = 0;

    for (int i = 0; i < weirdString.length; i += 1, wordIndex += 1) {
      if (normalString.charAt(i) == ' ') {
        weirdString[i] = " ";
        wordIndex = -1;
        continue;
      }

      if (wordIndex % 2 == 0) {
        weirdString[i]
            = Character.toString(normalString.charAt(i)).toUpperCase();
      }

      if (wordIndex % 2 == 1) {
        weirdString[i]
            = Character.toString(normalString.charAt(i)).toLowerCase();
      }
    }

    return String.join("", weirdString);
  }
}
